src/pages/products/Products.js

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import { FaTrashAlt, FaBatteryThreeQuarters, FaCheck, FaEdit, FaShoppingBag } from 'react-icons/fa';
+
+import { getProductsByCategory } from '../../services';
+import InputSpan from '../../components/form';
+import * as Constants from '../../constants';
+
+import './products.scss';
+
+class Products extends Component {
+
+    state = {
+        products: null,
+        displayFilter: []
+    };
+
+    insideRefs = [];
+
+    setRef = (numberOfRefs) => {
+        this.insideRefs = [...Array(numberOfRefs)]
+            .map((inputSpan) => inputSpan = React.createRef())
```

Comment:
why do you need refs? We use refs to implement some specific interaction with real DOM or in very rare case to use some react component methods

Reply:
@iliatcymbal Could you remind please, what is the better way for this feature?

Reply:
In this case I try re-use component, it the reason why I use "ref"

Reply:
I still can't understand why you need `ref` here. What the point to use it?